bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java

```diff
@@ -28,13 +28,15 @@
  *
  * @since 0.995.0
  */
-public class ErrorValue implements RefValue {
+public class ErrorValue extends RuntimeException implements RefValue {
 
-    private BType type;
-    private String reason;
-    private RefValue details;
+    private static final long serialVersionUID = 1L;
+    private final BType type;
+    private final String reason;
+    private final Object details;
```

Comment:
Details are modeled as bal records, right? Then shouldn't this be of MapValur type?

Reply:
Detail is actually ref value. So it can be any type.